src/world.cpp

```diff
@@ -2521,6 +2521,11 @@ void World_GenRoomCollision()
 {
     room_p r = global_world.rooms;
 
+    if(r == NULL)
+    {
+        return;
```

Comment:
Correct, except i feel that, it's good practice to check all pointers against NULL. Either way, if you have the room count set to 1 and global_world.rooms is actually null. That will cause a lot of problems, not saying it will happen but it's always a possibility. Perhaps it would have been better to use an assertion. I can revert this if you wish, let me know :)